packages/cli/test/integration/workflows/workflows.controller.ee.test.ts

```diff
@@ -1236,3 +1242,308 @@ describe('PATCH /workflows/:workflowId - activate workflow', () => {
 		expect(active).toBe(false);
 	});
 });
+
+describe('PUT /:workflowId/transfer', () => {
+	test('cannot transfer into the same project', async () => {
+		const destinationProject = await createTeamProject('Team Project', member);
+
+		const workflow = await createWorkflow({}, destinationProject);
+
+		await testServer
+			.authAgentFor(member)
+			.put(`/workflows/${workflow.id}/transfer`)
+			.send({ destinationProjectId: destinationProject.id })
+			.expect(400);
+	});
+
+	test('cannot transfer into a personal project', async () => {
+		const destinationProject = await createTeamProject('Team Project', member);
+
+		const workflow = await createWorkflow({}, destinationProject);
+
+		await testServer
+			.authAgentFor(member)
+			.put(`/workflows/${workflow.id}/transfer`)
+			.send({ destinationProjectId: memberPersonalProject.id })
+			.expect(400);
+	});
+
+	test('cannot transfer without workflow:move scope for the workflow', async () => {
+		const destinationProject = await createTeamProject('Team Project', member);
+
+		const workflow = await createWorkflow({}, anotherMember);
+
+		await testServer
+			.authAgentFor(member)
+			.put(`/workflows/${workflow.id}/transfer`)
+			.send({ destinationProjectId: destinationProject.id })
+			.expect(403);
+	});
+
+	test('cannot transfer without workflow:create scope in destination project', async () => {
+		const destinationProject = await createTeamProject('Team Project', anotherMember);
+
+		const workflow = await createWorkflow({}, member);
+
+		await testServer
+			.authAgentFor(member)
+			.put(`/workflows/${workflow.id}/transfer`)
+			.send({ destinationProjectId: destinationProject.id })
+			.expect(404);
+	});
+
+	test('project members cannot transfer workflows', async () => {
+		//
+		// ARRANGE
+		//
+		const sourceProject = await createTeamProject('Team Project 1');
+		await linkUserToProject(member, sourceProject, 'project:editor');
+		const destinationProject = await createTeamProject('Team Project 2', member);
+
+		const workflow = await createWorkflow({}, sourceProject);
+
+		//
+		// ACT & ASSERT
+		//
+		await testServer
+			.authAgentFor(member)
+			.put(`/workflows/${workflow.id}/transfer`)
+			.send({ destinationProjectId: destinationProject.id })
+			.expect(403);
+	});
+
+	test('can transfer from personal to team project', async () => {
```

Comment:
```suggestion
	test('can transfer from personal to team project and remove existing sharing', async () => {
```

Reply:
I think this is implicit behavior. A move will always sever all sharings, it's not optional. 

But I can split the tests to make one called: `moving from a personal project to a team project removes all existing sharings`.